protostar/git/git_repository.py

```diff
@@ -0,0 +1,262 @@
+import subprocess
+import re
+import os.path
+from dataclasses import dataclass
+from typing import Dict, Optional
+from pathlib import Path
+from protostar.git.git_exceptions import (
+    InvalidGitRepositoryException,
+    wrap_git_exception,
+)
+
+
+@dataclass
+class Submodule:
+    name: str
+    url: str
+    version: Optional[str] = None
```

Comment:
Nit: `Submodule::version` doesn't make sense. Consider `PackageInfo.from_submodule` to allow these two dataclasses to vary independently.

Reply:
I don't think it's particularly important at the moment. I might do it in a separate PR, perhaps related to package version enforcing.